semgrep-core/src/core/Rule.ml

```diff
@@ -167,15 +167,30 @@ type pformula = New of formula | Old of formula_old [@@deriving show, eq]
 (* Taint-specific types *)
 (*****************************************************************************)
 
-type sanitizer_spec = {
+type taint_source = {
+  formula : pformula;
+  label : string;  (** The label to attach to the data *)
+  requires : AST_generic.expr;
+      (** A Boolean formula over taint labels, the expression that
+       is being checked as a source must satisfy this in order to the
+       label to be produced. Note that with 'requires' a taint source
+       behaves a bit like a propagator ... *)
```

Comment:
Maybe you can add here that the expressions will be evaluated at matching time by Eval_generic?